src/invidious/videos/transcript.cr

```diff
@@ -0,0 +1,106 @@
+module Invidious::Videos
+  # Namespace for methods primarily relating to Transcripts
+  module Transcript
+    record TranscriptLine, start_ms : Time::Span, end_ms : Time::Span, line : String
+
+    def self.generate_param(video_id : String, language_code : String, auto_generated : Bool) : String
+      if !auto_generated
+        is_auto_generated = ""
+      elsif is_auto_generated = "asr"
+      end
+
+      object = {
+        "1:0:string" => video_id,
+
+        "2:base64" => {
+          "1:string" => is_auto_generated,
+          "2:string" => language_code,
+          "3:string" => "",
+        },
+
+        "3:varint" => 1_i64,
+        "5:string" => "engagement-panel-searchable-transcript-search-panel",
+        "6:varint" => 1_i64,
+        "7:varint" => 1_i64,
+        "8:varint" => 1_i64,
```

Comment:
I've noticed that varint 6 and 8 can fluctuate between 1 and 0 on the actual YouTube client's requests. It doesn't seem to do anything though and I can't deduce any patterns about it.